app/src/androidTest/java/com/github/houseorganizer/houseorganizer/CreateHouseholdActivityTest.java

```diff
@@ -75,15 +85,57 @@ public void setupHouseholds() throws ExecutionException, InterruptedException {
 
     @Test
     public void createHouseholdWorks() throws ExecutionException, InterruptedException {
+
         onView(withId(R.id.editTextHouseholdName)).perform(click(),
-                typeText("Test"), closeSoftKeyboard());
-        onView(withId(R.id.editTextLatitude)).perform(click(),
-                typeText("45"), closeSoftKeyboard());
-        onView(withId(R.id.editTextLongitude)).perform(click(),
-                typeText("45"), closeSoftKeyboard());
+                typeText(houseName), closeSoftKeyboard());
+        onView(withId(R.id.editTextAddress)).perform(
+                typeText("EPFL, Lausanne"), closeSoftKeyboard());
         onView(withId(R.id.submitHouseholdButton)).perform(click());
+        Thread.sleep(4000);
+        onView(withText(R.string.confirm)).perform(click());
+        Thread.sleep(1000);
+
+        Task<QuerySnapshot> t = db.collection("households").get();
+        Tasks.await(t);
+        List<DocumentSnapshot> house = t.getResult().getDocuments().stream()
+                .filter(doc -> doc.getString("name").equals(houseName)).collect(Collectors.toList());
+        assertEquals(house.size(), 1);
+    }
+
+    public void houseCreatedWithRightCoordinates() throws ExecutionException, InterruptedException {
+        onView(withId(R.id.editTextHouseholdName)).perform(click(),
+                typeText(houseName), closeSoftKeyboard());
+        onView(withId(R.id.editTextAddress)).perform(
+                typeText("EPFL, Lausanne"), closeSoftKeyboard());
+        onView(withId(R.id.submitHouseholdButton)).perform(click());
+        Map<String, Object> house = FirebaseTestsHelper.fetchHouseholdData(houseName, db);
+        assertEquals((Double)house.get("latitude"), 46.5, 0.1);
+        assertEquals((Double)house.get("longitude"), 6.5, 0.1);
+    }
+
+    @Test
+    public void emptyFieldsDoesNotCreateHouse() throws InterruptedException, ExecutionException {
+
+
+        Task<QuerySnapshot> t0 = db.collection("households").get();
+        onView(withId(R.id.submitHouseholdButton)).perform(click());
+
+        Task<QuerySnapshot> t1 = db.collection("households").get();
+
+        onView(withId(R.id.editTextAddress)).perform(typeText("address"), closeSoftKeyboard());
+        onView(withId(R.id.submitHouseholdButton)).perform(click());
```

Comment:
do you need line 122 ? since the second part of the if line 136 of CreateHouseholdActivity is not checked. Or you could also check when you input only the name and not the address.

Reply:
I do not really know how to remove text from the bar so I figured this tests enough do you think I should do this other case ?